Lib/fontTools/ttLib/tables/otTables.py

```diff
@@ -1297,6 +1298,144 @@ def preWrite(self, font):
 		return self.__dict__.copy()
 
 
+class ClipBox(getFormatSwitchingBaseTableClass("uint8")):
+
+	def as_tuple(self):
+		return tuple(getattr(self, conv.name) for conv in self.getConverters())
+
+	def __repr__(self):
+		return f"{self.__class__.__name__}{self.as_tuple()}"
+
+
+class ClipList(BaseTable):
+
+	def populateDefaults(self, propagator=None):
+		if not hasattr(self, "clips"):
+			self.clips = {}
+
+	def postRead(self, rawTable, font):
+		clips = {}
+		glyphOrder = font.getGlyphOrder()
+		for i, rec in enumerate(rawTable["ClipRecord"]):
+			rangesOverlap = False
+			for glyphID in range(rec.StartGlyphID, rec.EndGlyphID + 1):
+				try:
+					glyph = glyphOrder[glyphID]
+				except IndexError:
+					continue
+				if glyph not in clips:
+					clips[glyph] = copy.copy(rec.ClipBox)
+				else:
+					rangesOverlap = True
+			if rangesOverlap:
+				log.warning(
+					"ClipRecord %i overlap previous records; "
+					"redefined clip boxes are skipped",
+					i,
+				)
+		self.clips = clips
+
+	def groups(self):
+		glyphsByClip = defaultdict(list)
+		uniqueClips = {}
+		for glyphName, clipBox in self.clips.items():
+			key = hash(clipBox.as_tuple())
+			glyphsByClip[key].append(glyphName)
+			if key not in uniqueClips:
+				uniqueClips[key] = clipBox
+		return {
+			frozenset(glyphs): uniqueClips[key]
+			for key, glyphs in glyphsByClip.items()
+		}
+
+	def preWrite(self, font):
+		if not hasattr(self, "clips"):
+			self.clips = {}
+		clipBoxRanges = {}
+		glyphMap = font.getReverseGlyphMap()
+		for glyphs, clipBox in self.groups().items():
+			glyphIDs = sorted(
+				glyphMap[glyphName] for glyphName in glyphs
+				if glyphName in glyphMap
+			)
```

Comment:
aren't we hiding an error here? We don't need to fail or warn?

Reply:
I actually did this on purpose, so that if any of the base glyphs are removed in a subset operation the respective clipBox is also silently dropped. Alternatively I can have it blow up, and require whomever modifies the glyphOrder (renames or subsets) also update the COLR ClipList.

Reply:
other tables follow a similar pattern, e.g. hmtx.compile() method only loops over the ttFont.getGlyphOrder() (which may have changed since the table was decompiled/created), effectively skipping any glyph metrics whose glyphName is not in the current glyphOrder.